### PR TITLE
[Messaging] Necessary adjustments to allow using enum from MessageStore in the interfaces

### PR DESCRIPTION
**Pull request by VeithMetro**
@pwielders I also created a PR to ThunderInterfaces (https://github.com/rdkcentral/ThunderInterfaces/pull/266), but I'll have to rerun Actions there after this one is merged

### CODE REVIEW

File: Source/core/MessageStore.h
Comment (pwielders) [inline Source/core/MessageStore.h:1]: Why did you remove the move constructors/operators?

**Review by pwielders — DISMISSED** [review]

File: Source/core/MessageStore.h
Comment (VeithMetro) [reply Source/core/MessageStore.h:1]: `ProxyStubGenerator: ERROR: IMessageControl.h: /home/veith/buildroot/output/host/arm-buildroot-linux-gnueabihf/sysroot/usr/include/WPEFramework/core/MessageStore.h(59): parse error: rvalue references are not supported: 'Metadata &&'
CMake Error at /home/veith/buildroot/output/host/share/cmake/Modules/FindProxyStubGenerator.cmake:95 (message):
  ProxyStubGenerator generator failed.`
Sebastian said to remove it for now, and he is going to add it to the generator in the future.

**Comment by VeithMetro** [comment]
@pwielders looks like all of our Windows builds are crashing with such error:
`'SLEEPSLOT_POLLING_TIME': undeclared identifier`
And it is happening since this got merged: https://github.com/rdkcentral/Thunder/commit/a9e321ef1f5aeff320373ac0070fe36fc0449748#

File: Source/core/MessageStore.h
Comment (sebaszm) [reply Source/core/MessageStore.h:1]: As long as it does not serve any purpose? Does it not?

Comment (VeithMetro) [reply Source/core/MessageStore.h:1]: It does not, and it works as expected with move constructor/operator removed.
/edit: I've spoken with Sebastian and he suggested a proper way to test that, so I'll do that now to be extra sure.

**Comment by VeithMetro** [comment]
@pwielders I tested if the move constructor/operator is used anywhere, and it is not, but after talking to @sebaszm he proposed that instead of removing the move semantics, we can stop the parser before them - seems to be working as indented and this way we can keep the move constructor/operator

**Review by sebaszm — APPROVED** [review]
